community/modules/scripts/ramble-execute/templates/ramble_execute.tpl

```diff
@@ -0,0 +1,34 @@
+# Copyright 2023 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+- name: Execute Ramble Commands
+  hosts: localhost
+  vars:
+    spack_path: ${spack_path}
+    ramble_path: ${ramble_path}
+    log_file: ${log_file}
+  tasks:
+  - name: Execute ramble commands
+    ansible.builtin.shell: |
+      . {{ spack_path }}/share/spack/setup-env.sh
+      . {{ ramble_path }}/share/ramble/setup-env.sh
+      echo "" >> {{ log_file }}
+      echo " === Starting ramble commands ===" >> {{ log_file }}
+%{if length(COMMANDS) > 0 ~}
+%{for c in COMMANDS ~}
+      echo " === Executing command: ${c} === " >> {{ log_file }}
+      ${c} >> {{ log_file }}
```

Comment:
one of the frustrating bits about the Spack module is that it generates log entries _only_ in its own file and without the timestamps I need for debugging / alignment with other startup steps. Please do either of the following:

- stop redirecting output entirely and allow it to be captured with Ansible
- pipe to `tee -a`

Reply:
I've tested both of these solutions, and `ansible.builtin.shell` does not appear to print out stdout or stderr.

Supposedly, `ansible.builtin.command` can do this, but last time when we were looking at using command it was determined that it was strictly worse in terms of usability than shell.

Reply:
I added a comment about this above, but the solution I've pushed right now is to register the output and add `debug` to the ramble execute task.

Outside of this, I have been unable to get the output written to serial port 1 using `ansible.builtin.shell`.

Reply:
I believe this has been resolved.